src/storage/v2/inmemory/storage.cpp

```diff
@@ -118,6 +124,27 @@ auto FindEdges(const View view, EdgeTypeId edge_type, const VertexAccessor *from
                                                                  : to_vertex->InEdges(view, {edge_type}, from_vertex);
 }
 
+class PeriodicSnapshotObserver : public memgraph::utils::Observer<memgraph::utils::SchedulerInterval> {
+ public:
+  explicit PeriodicSnapshotObserver(memgraph::storage::InMemoryStorage &storage, memgraph::utils::Scheduler &scheduler)
+      : storage_{&storage}, scheduler_{&scheduler} {}
+
+  // String HAS to be a valid cron expr
+  void Update(const memgraph::utils::SchedulerInterval &in) override {
+    scheduler_->Setup(in);
+    scheduler_->SpinOne();
+    if (storage_->GetStorageMode() == StorageMode::IN_MEMORY_ANALYTICAL) {
+      scheduler_->Pause();
+    } else {
+      scheduler_->Resume();
```

Comment:
Why resume if it wasn't paused?

Reply:
This is leftover from a previous version where I would Pause/Resume based on the period defined.
This is now handled by the scheduler, so I don't actually need this here.
:+1: